modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,149 @@
+import { AddEventsBehaviour, AlloyComponent, AlloyEvents, AlloySpec, Behaviour, Button, Focusing, Input, Keying, Memento, NativeEvents, Representing } from '@ephox/alloy';
+import { Cell, Fun, Id, Optional } from '@ephox/katamari';
+import { Focus } from '@ephox/sugar';
+
+import Editor from 'tinymce/core/api/Editor';
+import { UiFactoryBackstage } from 'tinymce/themes/silver/backstage/Backstage';
+
+import { onControlAttached, onControlDetached } from '../../controls/Controls';
+import { updateMenuText, UpdateMenuTextEvent } from '../../dropdown/CommonDropdown';
+import { onSetupEvent } from '../ControlUtils';
+import { NumberInputSpec } from './FontSizeBespoke';
+
+interface BespokeSelectApi {
+  readonly getComponent: () => AlloyComponent;
+}
+
+const createBespokeNumberInput = (editor: Editor, _backstage: UiFactoryBackstage, spec: NumberInputSpec): AlloySpec => {
+  let currentComp: Optional<AlloyComponent> = Optional.none();
+
+  const getValueFromCurrentComp = (comp: Optional<AlloyComponent>): string =>
+    comp.map((alloyComp) => Representing.getValue(alloyComp)).getOr('');
+
+  const onSetup = onSetupEvent(editor, 'NodeChange', (api: BespokeSelectApi) => {
+    const comp = api.getComponent();
+    currentComp = Optional.some(comp);
+    spec.updateInputValue(comp);
+  });
+
+  const getApi = (comp: AlloyComponent): BespokeSelectApi => ({ getComponent: Fun.constant(comp) });
+  const editorOffCell = Cell(Fun.noop);
+
+  const customEvents = Id.generate('custom-number-input-events');
+
+  const isValidValue = (value: number): boolean => value >= 0;
+
+  const changeValue = (f: (v: number, step: number) => number): void => {
+    const text = getValueFromCurrentComp(currentComp);
+    const value = parseFloat(text.match(/(?<value>^[\d\.]+)/)?.groups?.value ?? '0');
+    const unitRegexp = new RegExp(`${value}(?<unit>\\D+)$`);
+    const unit = text.match(unitRegexp)?.groups?.unit ?? '';
+    const newValue = f(value, spec.getConfigFromUnit(unit).step);
+    const newValueWithUnit = `${isValidValue(newValue) ? newValue : value}${unit}`;
+
+    spec.onAction(newValueWithUnit);
```

Comment:
@spocke I tried to add a debounce, it works for the undo (just one), but the feedback in the editor contend is kinda strange because the effect is visible only at the end

Reply:
Yeah, that would be a bit strange but I guess there is no other way we can do this. Something needs to happen on the plus/minus and google docs is doing the same thing. So ignore my initial comment here.

Reply:
ok, reverted to the previous situation with multiple undo levels

Reply:
Maybe it makes sense to set a small amount of debounce time frame (aka 200-400ms) to skip intermediate clicks